api/v1/nodeinventory/service.go

```diff
@@ -0,0 +1,59 @@
+package nodeinventory
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	apiGRPC "github.com/stackrox/scanner/api/grpc"
+	v1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"github.com/stackrox/scanner/pkg/nodeinventory"
+	"google.golang.org/grpc"
+)
+
+// Service defines the node scanning service.
+type Service interface {
+	apiGRPC.APIService
+
+	v1.NodeInventoryServiceServer
+}
+
+// NewService returns the service for node scanning
+func NewService(nodeName string) Service {
```

Comment:
I usually return the impl-type from the constructor to allow the impl type to potentially satisfy many interfaces (also the future ones). Was there a reason to bind this constructor to the interface?

```suggestion
func NewService(nodeName string) serviceImpl {
```

Reply:
That's how all other gRPC services are defined. E.g. `api/v1/imagescan/service.go` or `api/v1/orchestratorscan/service.go`. 

I think the purpose of this abstraction is to have a single struct impl of the registered gRPC API. I'd prefer to keep it like this to comply with current abstraction.